quint/src/effects/base.ts

```diff
@@ -12,80 +12,78 @@
+import { Substitutions, applySubstitution, applySubstitutionToEntity, compose } from './substitutions'
 import { Error, ErrorTree, buildErrorLeaf, buildErrorTree } from '../errorTree'
 import { flattenUnions, simplify } from './simplification'
 import isEqual from 'lodash.isequal'
 
 /* Kinds for concrete effect components */
 export type ComponentKind = 'read' | 'update' | 'temporal'
 
-/* A concrete effect component, specifying a kind and the variables it acts upon */
-export interface EffectComponent { kind: ComponentKind, variables: Variables }
+/* A concrete effect component, specifying a kind and the entity it acts upon */
+export interface EffectComponent { kind: ComponentKind, entity: Entity }
 
-/* Concrete atomic effects specifying which kinds of effects act upon which variables */
+/* Concrete atomic effects specifying which kinds of effects act upon which entities */
 export interface ConcreteEffect { kind: 'concrete', components: EffectComponent[] }
 
 /* Arrow effects for expressions with effects depending on parameters */
 export interface ArrowEffect { kind: 'arrow', params: Effect[], result: Effect }
 
-/* A quantification variable, referring to an effect */
-export interface QuantifiedEffect { kind: 'quantified', name: string }
+/* A variable representing some effect */
+export interface EffectVariable { kind: 'variable', name: string }
 
 /*
- * The effect of a Quint expression, regarding which state variables are read
+ * The effect of a Quint expression, regarding which state entities are read
  * and/or updated
  */
 export type Effect =
   | ConcreteEffect
   | ArrowEffect
-  | QuantifiedEffect
+  | EffectVariable
 
 /*
- * An effect scheme, listing universally quantified variables (for names
- * refering to Variables) and effect variables (for names refering to Effects)
+ * An effect scheme, listing which effect variables (names referring to effects)
+ * and entity variables (names refering to entities) are universally quantified
+ * in the effect
  */
 export type EffectScheme = {
   /* The effect */
   effect: Effect,
   /* Universally quantified names refering to Effects */
   effectVariables: Set<string>,
-  /* Universally quantified names refering to Variables */
-  variables: Set<string>,
+  /* Universally quantified names refering to Entities */
+  entityVariables: Set<string>,
 }
 
-/* A state variable */
-export interface StateVariable {
+/* A state entity */
+export interface StateVariables {
   /* The variable name */
   name: string,
-  /* The variable id */
+  /* The id of the expression on which the state variable ocurred */
   reference: bigint
 }
 
 /*
- * The variables an effect acts upon. Either a list of state variables, a
- * quantification over them or a combination of other variables.
- * Uses the plural form since all forms represent a plurality of variables.
+ * The Entity an effect acts upon. Either a list of state variables, a
+ * variable or a combination of other entities.
```

Comment:
If one kind of "entity" is "a list of state variables", then it doesn't seem like a single  state variable can be a "state entity"? This seemes to confuse wholes and parts?

Reply:
The comment was suppose to say "state variable", not entity. This slipped through my reviews (auto-replacing was not trivial :stuck_out_tongue:), sorry